### PR TITLE
Only consider snaktype 'value' in entity.getlist (Fixes #18)

### LINKED ISSUE #18: entity.getlist(key) fails for "no value" entries
Example code:
```python
from wikidata.client import Client
client = Client()
hong_kong = client.get('Q8646')
locator_map_image = client.get('P242')
hong_kong.getlist(locator_map_image)
```
Error message:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lib/python3.6/site-packages/wikidata/entity.py", line 191, in getlist
    for snak in (claim['mainsnak'] for claim in claims)]
  File "/lib/python3.6/site-packages/wikidata/entity.py", line 191, in <listcomp>
    for snak in (claim['mainsnak'] for claim in claims)]
KeyError: 'datavalue'
```
I believe that this is due to the *no value* entry in the list for the [locator map image of Hong Kong](https://www.wikidata.org/wiki/Q8646#P242). In this case it appears to be an error in the underlying data, but the function should handle that gracefully. Since the docstring says "Return all values associated to the given key property in sequence.", such *no value* entries should probably simply get ignored.

### PR DESCRIPTION
**Pull request by Yorwba**
This implements my proposed solution for #18: *no value* entries are simply discarded.

### CODE REVIEW

File: wikidata/entity.py [hunk 1]
```diff
@@ -188,7 +188,8 @@ def getlist(self, key: 'Entity') -> Sequence[object]:
                          __import__('pprint').pformat(claims))
         decode = self.client.decode_datavalue
         return [decode(snak['datatype'], snak['datavalue'])
-                for snak in (claim['mainsnak'] for claim in claims)]
+                for snak in (claim['mainsnak'] for claim in claims)
+                if snak['snaktype'] == 'value']
```
Comment (dahlia) [inline wikidata/entity.py]: Could you add a test for this case?  See also other tests in *tests/entity_test.py* file.  You could run tests using [tox].

FYI you can make a fixture by fetching `Q8646`:

```bash
curl -o tests/fixtures/entities/Q8646.json \
  https://www.wikidata.org/wiki/Special:EntityData/Q8646.json
```

[tox]: https://tox.readthedocs.io/

Comment (Yorwba) [reply wikidata/entity.py]: I added the test case you requested. I verified that it fails before, but succeeds after this change.

**Comment by Yorwba** [comment]
I amended the commit to include a changelog.

**Comment by dahlia** [comment]
Thank you!

**Pull request merged**